lib/wcgi-runner/src/module_loader/cached.rs

```diff
@@ -0,0 +1,48 @@
+use tokio::sync::Mutex;
+
+use crate::{
+    module_loader::{ModuleLoader, ModuleLoaderContext},
+    Error,
+};
+
+use super::LoadedModule;
+
+pub(crate) struct Cached<L> {
+    loader: L,
+    invalidated: Box<dyn Fn() -> bool + Send + Sync>,
+    cached: Mutex<Option<LoadedModule>>,
+}
+
+impl<L> Cached<L> {
+    pub(crate) fn new(loader: L, invalidated: impl Fn() -> bool + Send + Sync + 'static) -> Self {
+        Self {
+            loader,
+            invalidated: Box::new(invalidated),
+            cached: Mutex::new(None),
+        }
+    }
+}
+
+#[async_trait::async_trait]
+impl<L: ModuleLoader> ModuleLoader for Cached<L> {
+    async fn load(&self, ctx: ModuleLoaderContext<'_>) -> Result<LoadedModule, Error> {
+        let mut cached = self.cached.lock().await;
+
```

Comment:
Why do we need this exactly? We already have the `wasmer-cache` crate that should handle this

Reply:
Regardless of whether we use a `Mutex<Option<LoadedModule>>` or a `wasmer_cache::FileSystemCache`, we still need to have some sort of struct which can be used as a `ModuleLoader`.

I used a mutex-based solution just because it's trivial to write and was good enough to get things implemented.

Reply:
But the `wasmer_cache::FileSystemCache` is already provided by the CLI, right?
I don't think the caching mechanism should be defined in the runner (but plugged into it).

I think we might be doing an early optimization by having the `wcgi-runner` standalone instead of being first integrated on the cli first (which I think might be simpler)